packages/datadog-instrumentations/src/fastify.js

```diff
@@ -126,13 +127,22 @@ function preValidation (request, reply, done) {
     }
 
     if (bodyParserReadCh.hasSubscribers && request.body) {
-      const abortController = new AbortController()
-
       bodyParserReadCh.publish({ req, res, body: request.body, abortController })
 
       if (abortController.signal.aborted) return
     }
 
+    if (pathParamsReadCh.hasSubscribers && request.params) {
+      pathParamsReadCh.publish({
+        req,
+        res,
+        abortController,
```

Comment:
if you do this for the 3 different calls, you don't create it until it's needed, but you also cache it for the others. Wdyt ?
```suggestion
        abortController: abortController ?? abortController = new AbortController(),
```